### PR TITLE
Implement all names and units

### PR DESCRIPTION
**Pull request by rcaneill**
This is WIP, but I submit the PR so you can start to check the functions I already did.
This is long to do, but I'm soon there!

### CODE REVIEW

**Comment by rcaneill** [comment]
I finished adding the units, standard names (a minority of the returned variables have a standard name), and names for the variables.

Some of the functions had some weird units in the matlab / python docs (specially the derivatives with enthalpy), could you double check what I did?

Don't hesitate to commit any minor typo / correction.

**Comment by rcaneill** [comment]
I removed the standard names of sigma1, sigma2, etc while #32 is not solved.
I think that we can merge and release v0.2.0 without the standard_name for these functions, and solve issue #32 for v0.2.1

File: gsw_xarray/_names.py [hunk 1]
```diff
@@ -21,17 +40,148 @@
...
+    "distance": "distance",
+    "dynamic_enthalpy": "dynamic_enthalpy",
+    "enthalpy": "enthalpy",
+    "enthalpy_CT_exact": "enthalpy",
+    "enthalpy_diff": "enthalpy",
+    "enthalpy_first_derivatives": ("h_SA", "h_CT"),
+    "enthalpy_first_derivatives_CT_exact": (
+        "h_SA",
+        "h_CT",
+    ),  # difference with matlab code that return 3 outputs
```
Comment (DocOtak) [inline gsw_xarray/_names.py:69]: Seems the docs are wrong:
https://github.com/TEOS-10/GSW-Matlab/issues/7

**Comment by DocOtak** [comment]
I think this is looking pretty good. Thanks for working on it.

Nothing is standing out to me as blocking. If you think it's ready, go ahead and merge.

**Pull request merged**